src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/AllocFreeConcurrentStack.cs

```diff
@@ -57,19 +77,15 @@ private static Stack<RefAsValueType<T>> ThreadLocalStack
                 return (Stack<RefAsValueType<T>>)stackObj;
             }
         }
+#endif
     }
 
+#if !NETCOREAPP
     internal static class AllocFreeConcurrentStack
     {
-        // WARNING: We allow diagnostic tools to directly inspect this member (t_stacks).
-        // See https://github.com/dotnet/corert/blob/master/Documentation/design-docs/diagnostics/diagnostics-tools-contract.md for more details.
-        // Please do not change the type, the name, or the semantic usage of this member without understanding the implication for tools.
-        // Get in touch with the diagnostics team if you have questions.
-
         // Workaround for https://github.com/dotnet/runtime/issues/4731.
-        // When that's fixed, a [ThreadStatic] Stack should be added back to AllocFreeConcurrentStack<T>.
```

Comment:
This bug has been fixed a long time ago. I have kept the workaround for netstandard builds.